### PR TITLE
Properly set executable bits when untar-ing ScanCode

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/991)

### CODE REVIEW

File: utils/src/main/kotlin/ArchiveUtils.kt [hunk 1]
```diff
@@ -83,6 +83,15 @@ fun File.unpackTar(targetDirectory: File) {
             target.outputStream().use { output ->
                 it.copyTo(output)
             }
+
+            if (!OS.isWindows) {
+                // Note: In contrast to Java, Kotlin does not support octal literals, see
+                // https://kotlinlang.org/docs/reference/basic-types.html#literal-constants.
+                // The bit-triplets from left to right stand for user, groups, other, respectively.
+                if (entry.mode and 0b001_000_001 != 0) {
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply utils/src/main/kotlin/ArchiveUtils.kt]: There's a difference between `TarArchiveEntry` and `ZipArchiveEntry`: In the former it's just called `mode`, in the latter it`s called `unixMode`.

File: scanner/src/main/kotlin/scanners/ScanCode.kt [hunk 2]
```diff
@@ -135,10 +135,12 @@ class ScanCode(config: ScannerConfiguration) : LocalScanner(config) {
     }
 
     override fun bootstrap(): File {
-        // Use the .zip file despite it being slightly larger than the .tar.bz2 file as the latter for some reason does
-        // not complete to unpack on Windows.
-        val url = "https://github.com/nexB/scancode-toolkit/releases/download/v$scannerVersion/" +
-                "scancode-toolkit-$scannerVersion.zip"
+        val archive = when {
+            OS.isWindows -> "scancode-toolkit-$scannerVersion.zip"
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Did you maybe review an older version? I already had added such a comment.

File: utils/src/main/kotlin/ArchiveUtils.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply utils/src/main/kotlin/ArchiveUtils.kt]: Oh ok. all fine then, thanks!

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: yes, it seems it was the github UI.

**Pull request merged**